src/segment_strava.go

```diff
@@ -0,0 +1,258 @@
+	StartDate time.Time `json:"start_date"`
+	Name      string    `json:"name"`
+	Distance  float64   `json:"distance"`
+	Duration  float64   `json:"moving_time"`
+}
+
+type TokenExchange struct {
+	TokenType    string `json:"token_type"`
+	AccessToken  string `json:"access_token"`
+	RefreshToken string `json:"refresh_token"`
+	ExpiresIn    int    `json:"expires_in"`
+	ExpiresAt    int    `json:"expires_at"`
+}
+
+func (ns *strava) enabled() bool {
+	data, err := ns.getResult()
+	if err != nil {
+		return false
+	}
+	ns.StravaData = *data
+	ns.ActivityIcon = ns.getActivityIcon()
+	ns.Hours = ns.getHours()
+	ns.Ago = ns.getAgo()
+
+	return true
+}
+
+func (ns *strava) getHours() int {
+	hours := time.Since(ns.StartDate).Hours()
+
+	return int(math.Floor(hours))
+}
+
+func (ns *strava) getAgo() string {
+	if ns.Hours > 24 {
+		days := int32(math.Floor(float64(ns.Hours) / float64(24)))
+		return fmt.Sprintf("%d", days) + string('d')
+	}
+	return fmt.Sprintf("%d", ns.Hours) + string("h")
+}
+
+func (ns *strava) getActivityIcon() string {
+	switch ns.Type {
+	case "Ride":
+		return ns.props.getString(RideIcon, "\uf5a2")
+	case "Run":
+		return ns.props.getString(RunIcon, "\ufc0c")
+	case "NordicSki":
+	case "AlpineSki":
+	case "BackcountrySki":
+		return ns.props.getString(SkiingIcon, "\ue213")
+	case "WorkOut":
+		return ns.props.getString(WorkOutIcon, "\ue213")
+	default:
+		return ns.props.getString(UnknownActivityIcon, "\ue213")
+	}
+	return ns.props.getString(UnknownActivityIcon, "\ue213")
+}
+
+func (ns *strava) string() string {
+	segmentTemplate := ns.props.getString(SegmentTemplate, "{{.Ago}}")
+	template := &textTemplate{
+		Template: segmentTemplate,
+		Context:  ns,
+		Env:      ns.env,
+	}
+	text, err := template.render()
+	if err != nil {
+		return err.Error()
+	}
+
+	return text
+}
+
+func (ns *strava) getAccessToken() (string, error) {
+	var at, rt string
+	var hasAt, hasRt bool
+	var err error
+
+	// get directly from cache
+	at, hasAt = ns.env.cache().get(StravaAccessToken)
+	if !hasAt {
+		rt, hasRt = ns.env.cache().get(StravaRefreshToken)
+		if hasRt {
+			// use refreshtoken stored in cache to get a new
+			at = "GET FROM OHMYPOSH API OR USE FUNCTION BELOW"
+			// at, err = ns.getNewAccessToken(rt) // ToDo: Replace to with api ohmyposh.dev to get new tokens
+		} else {
+			// use initial tokens from property
+			at = ns.props.getString(InitialAccessToken, "")
+			rt = ns.props.getString(InitialRefreshToken, "")
+			if rt == "" || at == "" {
+				return "", errors.New("initial tokens not set")
+			}
+			// request new access token to verify valid refresh token
+			// at, err = ns.getNewAccessToken(rt) // ToDo: Replace to with api ohmyposh.dev to get new tokens
+		}
+		if err != nil {
+			return "", err
+		}
+	}
+	return at, err
+}
+
+// This method will be implemented in ohmyposh.dev api (I guess it works, just neet to set StravaClientSecret)
+func (ns *strava) getNewAccessToken(refreshToken string) (string, error) {
```

Comment:
would rename to refreshToken(), we can also get the `refresh_token` here rather than provide it